src/cache.rs

```diff
@@ -36,8 +36,8 @@ impl Cache {
         F: Future<Output = Result<JsonWebKeySet, Error>> + Send + 'static,
     {
         let read: RwLockReadGuard<Entry> = self.inner.read().await;
-        let is_entry_expired: bool = (*read).is_expired();
-        let get_key_result: Result<JsonWebKey, Error> = (*read).set.get_key(key).cloned();
+        let is_entry_expired: bool = read.is_expired();
```

Comment:
Those changes will ends up with merge conflict with #12

Reply:
yeah I know, I needed to upgrade rust here too and clippy complains about those even in this branch, i will rebase that eventually